File: LeetCode/C/83. Remove Duplicates from Sorted List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteDuplicates(struct ListNode* head){
    struct ListNode* k=head;
    while(k)
    {
        while( k->next && k->val==k->next->val)
            k->next=k->next->next;
        k=k->next;
    }
    return head;
}

File: LeetCode/Java/83. Remove Duplicates from Sorted List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode k = head;
        while (k!= null)
        {
            while(k.next!= null && k.val==k.next.val)
                k.next=k.next.next;
            k=k.next;
        }
        return head;
    }
}